flycheck-haskell.el

```diff
@@ -137,7 +146,10 @@ Take the base command from `flycheck-haskell-runghc-command'."
 
 (defun flycheck-haskell--read-configuration-with-helper (command)
   (with-temp-buffer
-    (pcase (apply 'call-process (car command) nil t nil (cdr command))
+    ;; Discard stderr. When runghc loads a package environment it prints some
+    ;; info to the stderr. Apparently there is no GHC parameter to silence that
+    ;; message.
+    (pcase (apply 'call-process (car command) nil (list t nil) nil (cdr command))
```

Comment:
This discards the whole stderr, so if there is a fatal error, it may make things more difficult to understand.

According to [`call-process`](https://www.gnu.org/software/emacs/manual/html_node/elisp/Synchronous-Processes.html) API documentation there is no direct way to put the stderr into a buffer.

> You can't directly specify a buffer to put the error output in; that is too difficult to implement. But you can achieve this result by sending the error output to a temporary file and then inserting the file into a buffer when the subprocess finishes.

So I'm not sure how to go about it or whether it's worth the trouble.

Do you know any examples dealing with this problem? It sounds like this might usual enough for a solution with some idiomatic code snippet.

Reply:
Discarding stderr may indeed hide some errors and make debugging much more difficult. From which version `runghc` starts printing loading messages? It doesn't seem to do that on CI which tests versions up to 8.4.3.

Reply:
No, I don't think that asynchronous process will be an issue. But we'll likely have to black until it finishes.